sysdump/sysdump.go

```diff
@@ -437,11 +439,13 @@ func (c *Collector) Run() error {
 			Description: "Collecting the Cilium daemonset",
 			Quick:       true,
 			Task: func(ctx context.Context) error {
-				v, err := c.Client.GetDaemonSet(ctx, c.Options.CiliumNamespace, ciliumDaemonSetName, metav1.GetOptions{})
+				v, err := c.Client.ListDaemonSet(ctx, c.Options.CiliumNamespace, metav1.ListOptions{
+					LabelSelector: c.Options.CiliumDaemonSetSelector,
+				})
```

Comment:
Given that this will return the list of daemonsets, it will change the format of the daemonset file currently from a single daemonset to a list. Since we know only the Cilium daemonset is supposed to be selected with the label selector, does it make sense to always grab the first item from the list to maintain the current file format of a single daemonset instead of a list. Of course, we'll need to check if the list is empty before we index into the list.

Reply:
thanks, making sense to me 💯